doc/reflection-milestone2.md

```diff
@@ -1,43 +1,46 @@
-that we believe will contribute to employee attrition based on IBM
-employee satisfaction survey data. We also implemented 2 dropdown
-filters (`Department` and `Gender`) and 1 slide bar (`Age`) for the
-management team to play with in order to see whether the attrition rate
-vary between features. By filtering the data, our plots show the
-proportion of attrition among the 4 key factors, which could help the
-management team to make better decisions.
+that contributes to employee attrition based on IBM
+employee satisfaction survey data. We also implemented a total of 
+3 interactive widgets to our app, 2 dropdown filters (`Department` and `Gender`) 
+and 1 slide bar (`Age`) for the management team to play with in order to see 
+whether the attrition rate vary between features. By filtering the data, our plots 
+show the attrition among the 4 key factors among different employee segment, which could help the
+management team to develop their action plan for next step.
 
-There are two things we have in mind but not yet implemented
-successfully. We have four widgets in our proposal but we are only able
-to implement three of them. We left out `Job Role` feature because this
-feature has interaction with `Department` feature. The selected job role
-has to be included in the selected department for the dashboard to have
-data to plot, otherwise we could only get empty plots. Therefore, in
-this milestone, we dropped `Job Role` and only kept `Department` in the
+We have four widgets in our proposal but we decided to only 
+implement three of them. We left out `Job Role` feature because this
+feature is depending on `Department` feature. The selected job role
+needs to be belong to the selected department for the dashboard to have
+data to plot, otherwise we could only get empty plots. Also, the sample size for 
+specific roles is quite small and regrouping them may just become another department oriented variable. 
+Therefore, in this milestone, we dropped `Job Role` and only kept `Department` in the
 filter session. Another thing we have not implemented is that for now
-our default plots is showing the attrition based on female in sales
+our default plots are showing the attrition based on female in sales
 department with age between 18 to 60. We would like to make the default
 plots to show the attrition proportion among all the data instead of
-filtered data. We are hoping to implement those two aspects in the
-future milestone.
+filtered data. We are hoping to implement this in the
+future milestone. We also identified an issue with altair plotting 
+the ordinal categorical variable in alphabetical order. Changing it
+with domain attribute of of the plot scale is having conflict with Dash. 
+Now we have relabeled them with number in order to show the desired order,
+we will try to figure out a better way in the next 2 milestones. 
 
 We believe our dashboard provide a general picture of how attrition rate
```

Comment:
For: "We believe our dashboard provide a general picture of how attrition rate varies among different features based on the 4 key factors we observed."

Suggestion to consider: "Our user-interactive dashboard demonstrates a comprehensive overview of how employee's attrition distributes in four key aspects."

Reply:
good point and will restructure the next few sentences a bit.